spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +113,43 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+    table: DeltaTableV2,
+    target: String,
+    create: Boolean,
+    replace: Boolean,
+    ifNotExists: Boolean,
+    tableProperties: Map[String, String],
+    versionAsOf: Option[Long] = None,
+    timestampAsOf: Option[String] = None
+  ): DataFrame = withActiveSession(sparkSession) {
+    val (targetRelation, targetPath, targetIdentifier) = if (new Path(target).isAbsolute()) {
```

Comment:
The way how we create `targetIdentifier` can be simplified because we have `getTableIdentifierIfExists` - https://github.com/delta-io/delta/blob/18eb1a6cbd09049eb835586a4d607fbd9673a1ee/spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaTableV2.scala#L98, happy to make the change for you if you agree.

Reply:
We only have a DeltaTableV2 for the source, not the target

Reply:
`val source = DataSourceV2Relation.create(table, None, targetIdentifier.map(
      id => Identifier.of(id.database.toArray, id.table)))`
      
when `targetIdentifier` is `sparkSession
        .sessionState
        .sqlParser
        .parseTableIdentifier(target)` or `None` looks kinda strange to me, shouldn't this be

```
val sourceIdentifier = table.getTableIdentifierIfExists.map(id =>
  Identifier.of(id.database.toArray, id.table))
```
        
`val source = DataSourceV2Relation.create(table, None, sourceIdentifier.map(
      id => Identifier.of(id.database.toArray, id.table)))`
      
 We want to create the source relation right?

Reply:
Oh you mean the line below, yeah thats not right that shouldn't be the target identifier

Reply:
Fixed